internal/operands/tekton-tasks/reconcile.go

```diff
@@ -307,3 +276,21 @@ func getTektonTasksNamespace(request *common.Request) string {
 	}
 	return request.Instance.Namespace
 }
+
+func filterResources[PT interface {
```

Comment:
Appreciate use of generics! Can this be simplified to just use the `metav1.Object` interface? It is working right now but looks very complicated.

Reply:
Yes, I agree that's very complicated, also there is no Go docs for it, but it works. I tried to do it with just `metav1.Object`, I couldn't make it work. @akrejcir suggested to do that way yesterday https://github.com/kubevirt/ssp-operator/pull/532#discussion_r1191019175.

Reply:
@akrejcir's yesterday's explanation for it:
```
If you would define simply [T metav1.Object] it means that parameter T is a type that implements interface metav1.Object . Which means it is a pointer to a struct, for example T = *v1.Pod.
When you have : []T it is in fact []*v1.Pod. So when a method accepts []T parameter, it does not accept []v1.Pod to it.
The PT interface {*T ; meta1.Object } is a trick. It means that type PT satisfies an anonymous typeset defined directly in the function signature. This typeset interface { *T; metav1.Object } means that it is a pointer to T and implements interface metav1.Object.  Which means that for example PT = *v1.Pod .
It would be nice if golang compiler was smarter and we didn't need PT. :)
```

Reply:
Better explanation: https://stackoverflow.com/a/71444968

Reply:
IMO we should not merge this just because it works.

I see the pain in changing the slices to hold pointers instead. Is there another maybe less 'generic' but still prettier solution?

Reply:
What about something casting to keep this simple?

```go
func filterResources[T any](items []T, trimSuffix bool) []T {
	var results []T
	for _, obj := range items {
		metaObj, _ := any(obj).(metav1.ObjectMeta)
		name := metaObj.GetName()
		if trimSuffix {
			name = strings.TrimSuffix(name, "-task")
		}
		if _, ok := AllowedTasks[name]; ok {
			results = append(results, obj)
		}
	}
	return results
}
```

Reply:
Ah sweet, thanks @lyarwood. Let's use it then.